src/components/CreatePost.js

```diff
@@ -44,9 +42,11 @@ export default function PublishPost() {
     }
     publishPost(form)
       .then(() => {
-        // TODO:Incluir a função para mostrar todos os posts
-        // showPosts();
         setDisabled(false);
+        setForm({
+          url: "",
+          description: "",
+        });
       })
       .catch((answer) => {
         alert("Houve um erro ao publicar seu link");
```

Comment:
I have a doubt. Wouldn't it be necessary to add the useEffect hook to call the 'publishPost' function?

Reply:
> I have a doubt. Wouldn't it be necessary to add the useEffect hook to call the 'publishPost' function?

I don't think so, Henrique. Use Effect would be necessary on a function to get data that would be rendered by React, to prevent an infinite rendering loop on the component. This functions is called by a submit event and is a post requisition only.

We should add a refresh state variable for the publishing of a new post to trigger a new render of the posts array(not implemented yet)

Reply:
Yes, you're right